unsafe-fields/src/lib.rs

```diff
@@ -220,6 +225,26 @@ impl<O: ?Sized, F: ?Sized, const NAME_HASH: u128> Unsafe<O, F, { NAME_HASH }> {
         &self.field
     }
 
+    /// Gets a reference to the inner value safely so long as the inner value is
+    /// immutable.
+    ///
+    /// If [`F: Immutable`][immutable], then `F` does not permit interior
+    /// mutation, and so it is safe to return a reference to it.
+    ///
+    /// [immutable]: zerocopy_0_8::Immutable
+    #[inline(always)]
+    #[cfg(feature = "zerocopy_0_8")]
+    #[cfg_attr(doc_cfg, doc(cfg(feature = "zerocopy_0_8")))]
+    pub const fn as_ref_checked(&self) -> &F
```

Comment:
My preference would be `as_ref_unchecked` for the unsafe method, and make the safe thing the shortest to type (`.as_ref`). Most types are `Immutable`, so I additionally expect the safe version will be the common case!

Reply:
Feels weird to have most of the API suffixed with `_unchecked`, but I agree that that's philosophically the right thing to do.

Done.